qiskit/ignis/experiments/calibration/instruction_data/gate_and_schedules_definition.py

```diff
@@ -0,0 +1,203 @@
+from qiskit.pulse.channels import PulseChannel
+from qiskit.providers.basebackend import BaseBackend
+
+from qiskit.ignis.experiments.calibration.instruction_data.database import PulseTable
+
+
+class InstructionsDefinition:
+    """
+    Class to track calibrated instructions for the calibration module of Ignis.
+
+    The class allows users to define single-pulse schedules as quantum circuits and
+    then leverage the quantum circuit compose operation to compose more complex
+    circuits from basic circuits.
+    """
+
+    def __init__(self, backend: BaseBackend):
+        """
+        Args:
+            backend: The backend for which the instructions are defined.
+        """
+
+        # Dict to store the instructions we are calibrating
+        self._instructions = {}
+
+        # Table to store the calibrated pulse parameters
+        self._pulse_table = PulseTable()
+
+    @property
+    def instructions(self) -> Dict[Tuple[str, Iterable], QuantumCircuit]:
+        """Return the instructions as a dict."""
+        return self._instructions
+
+    def get_calibration(self):
+        """Returns the calibrations."""
+        return self._pulse_table.filter_data()
+
+    def get_instruction_template(self, name: str, qubits: Iterable) -> QuantumCircuit:
+        """
+        Args:
+            name: Name of the instruction to get.
+            qubits: Set of qubits to which the instruction applies.
+
+        Returns:
+            A QuantumCircuit with parameters in it.
+        """
+        return self._instructions.get((name, qubits), None)
+
+    def get_instruction(self, name: str, qubits: Iterable,
+                        free_parameters: List = None) -> QuantumCircuit:
+        """
+        Returns the QuantumCircuit of the instruction where all parameters aside for those
+        explicitly specified are bound to their calibration.
+
+        Args:
+            name: name of the instruction to retrive
+            qubits: qubits to which the instruction applies.
+            free_parameters: Parameters that should be left unbound. If None is specified then
+                all parameters will be bound to their calibrated values.
+        """
+        circ = self.get_instruction_template(name, qubits)
+
+        # TODO Get their values
+        binding_dict = {}
+        for param in circ.parameters:
+            binding_dict[param] = self._get_parameter_value(param)
+
+        return circ.assign_parameters(binding_dict)
+
+    def _get_parameter_value(self, parameter: ParameterExpression) -> Union[float, int, complex]:
+        """
+        Helper method.
+
+        TODO This could be simplified by better integrating with PulseTable
+        """
+        inst_name, channel, name = parameter.name.split('.')
+        qubits = self._get_qubits(channel)
+
+        return list(self._pulse_table.get_parameter(qubits, channel, inst_name, '', 1.0, name).values())[0].value
```

Comment:
I think you should define pulse type. The function call this helper should know what the pulse type is, so we can add pulse_type entry to input args, or you can add pulse type info to your parameter name. Otherwise we may return wrong data if two pulse types exist in the database.

Reply:
Yes, this is temporary. Here, we face a design choice. Currently, we can support multiple pulse shapes for a given pulse (i.e. pulse table can hold simultaneously `x` as `Drag` and `Gaussian`). It's unclear if we want to allow this. I.e. should we restrict the pulse table to hold only one pulse shape per pulse?